sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala

```diff
@@ -508,8 +509,22 @@ object UnsupportedOperationChecker extends Logging {
         case Sample(_, _, _, _, child) if child.isStreaming =>
           throwError("Sampling is not supported on streaming DataFrames/Datasets")
 
-        case Window(_, _, _, child) if child.isStreaming =>
-          throwError("Non-time-based windows are not supported on streaming DataFrames/Datasets")
+        case Window(windowExpression, _, _, child) if child.isStreaming =>
+          val windowFuncs = windowExpression.flatMap { e =>
+            e.collect {
+              case we: WindowExpression =>
+                s"'${we.windowFunction}' as column '${e.toAttribute.sql}'"
```

Comment:
Could you change quoting of the expression and the id using `toSQLExpr` and `toSQLId`

Reply:
and just pass both to `nonTimeWindowNotSupportedInStreamingError`, and put `as column` to error-classes.json.